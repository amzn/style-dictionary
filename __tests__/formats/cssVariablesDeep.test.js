var fs = require('fs-extra');
var helpers = require('../__helpers');
const formats = require('../../lib/common/formats');

var file = {
  "destination": "__output/",
  "format": "css/variables-deep"
};

var dictionary = {
  "allProperties": [
    {
      name: "color-blue-20",
      value: "#0000ff",
      original: {
        value: "#0000ff",
      },
    },
    {
      name: "primary",
      value: "#00ff00",
      original: {
        value: "{color.blue-20.value}",
      },
    },
    {
      name: "icon",
      value: "acme-icon",
      attributes: { "data-type": "string" },
    },
    {
      name: "width",
      value: "100px",
      attributes: { "data-type": "number" },
    }
  ]
};

describe('formats', () => {
  describe('css/variables-deep', () => {

    // mock the Date.now() call to a fixed value
    const constantDate = new Date('2000-01-01');
    const globalDate = global.Date;

    var formatter = formats['css/variables-deep'].bind(file);

    beforeEach(() => {
      global.Date = function() { return constantDate };
      helpers.clearOutput();
    });

    afterEach(() => {
      global.Date = globalDate;
      helpers.clearOutput();
    });

    it('should generate a css variables file with alias references', () => {
      fs.writeFileSync('./__tests__/__output/output.css', formatter(dictionary, {}) );
      const testFile = fs.readFileSync("./__tests__/__output/output.css", "UTF-8");
      expect(testFile).toMatchSnapshot();
    });

    it('should work with a prefix', () => {
      fs.writeFileSync('./__tests__/__output/output.css', formatter(dictionary, { prefix: "acme" }) );
      const testFile = fs.readFileSync("./__tests__/__output/output.css", "UTF-8");
      expect(testFile).toMatchSnapshot();
    });
  });

});
