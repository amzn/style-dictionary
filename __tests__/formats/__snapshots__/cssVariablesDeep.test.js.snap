// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formats css/variables-deep should generate a css variables file with alias references 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

:root {
	--color-blue-20: #0000ff;
	--primary: var(--color-blue-20);
	--icon: \\"acme-icon\\";
	--width: 100px
}
"
`;

exports[`formats css/variables-deep should work with a prefix 1`] = `
"/**
 * Do not edit directly
 * Generated on Sat, 01 Jan 2000 00:00:00 GMT
 */

:root {
	--color-blue-20: #0000ff;
	--primary: var(--acme-color-blue-20);
	--icon: \\"acme-icon\\";
	--width: 100px
}
"
`;
